Add instructions for stewarding a release (#89)
conda-forge/modin-feedstock: HOWTO.md

# Updating the Conda Feedstock

> Once the new release is finalized, a bot will open a PR on this repository in order to update the conda feedstock. CI will probably fail on this PR, because there are patches that we need to manually update. These patches remove dependencies from `env_hdk.yml` and `setup.py` in the Modin repository. You'll need to update the dependencies in `recipe/meta.yaml` based off of the dependencies we remove in the patches.

Reviewer: ```suggestion

Once the new release is finalized, a bot will open a PR on this repository in order to update the conda feedstock. CI will probably fail on this PR, because there are patches that we need to manually update. These patches remove dependencies from `env_hdk.yml` and `setup.py` in the Modin repository. You'll need to update the dependencies in `recipe/meta.yaml` based off of the dependencies we remove in the patches.
```

> You'll need to make these changes to your fork of the modin-feedstock repo. An ideal way to start is to add the [bot's fork](https://github.com/regro-cf-autotick-bot/modin-feedstock.git) as a remote, and work off of its branch for the release, and once you're done, close the bot's PR and open your own PR with your new branch.

Reviewer 2: There is an option to just push to the bot's branch if you're a maintainer of the repo.

Also, closing the bot's PR shouldn't be necessary, I think it would auto-close if someone merges a PR which does the same version update as the bot intended to do.

Reviewer: Agree, the docs should reflect this.

> ## Creating Commits for Patches

Reviewer: ```suggestion
## Creating Commits for Patches

```

Inside either your fork of Modin, or the release copy you cloned in order to build the pip package, checkout either:
* The release branch (if working on releasing a patch version)
* Modin's master branch (if working on a major or a minor version)

Once you've checked out these files, you need to make 2 commits (can be made in any order).

> ### `env_hdk.yml` commit

Reviewer: ```suggestion

### `env_hdk.yml` commit

```

You'll need to remove the `pandas`, `numpy`, `pyhdk` and `s3fs` dependencies from this file. In order to ensure that you have removed all of the necessary dependencies,
check the patch introduced by the PR for the previous release version. After making these changes, commit the file with the commit message "Remove dependencies from env_hdk.yml."

> ### `setup.py` commit

Reviewer: ```suggestion

### `setup.py` commit

```

You'll need to replace the following list entries with empty lists in `setup.py`:
* `dask_deps`
* `ray_deps`
* `unidist_deps`
* `spreadsheet_deps`
* `sql_deps`
* `install_requires`

Once you've set these entries to empty lists, commit the file with the commit message "Remove dependencies from setup.py."

> ## Building Patches

Reviewer: ```suggestion
## Building Patches

```

Once you've committed the above two commits, run `git format-patch -2 HEAD`. This will output two filenames - each corresponding to one of the commits above. Replace each patch in the feedstock repo with the corresponding patch generated by this command.

> ## Updating `recipe/meta.yaml`

Reviewer: ```suggestion
## Updating `recipe/meta.yaml`

```

You can either push directly to the bot's PR in order to make these changes, or you can make these changes to your fork of the modin-feedstock repo. If you are using your fork of the modin-feedstock repo, an ideal way to start is to add the [bot's fork](https://github.com/regro-cf-autotick-bot/modin-feedstock.git) as a remote, and work off of its branch for the release - opening your own PR when you're done.

## Creating Commits for Patches

Inside either your fork of Modin, or the release copy you cloned in order to build the pip package, checkout either:
* The release branch (if working on releasing a patch version)
* Modin's master branch (if working on a major or a minor version)

Once you've checked out these files, you need to make 2 commits (can be made in any order).

### `env_hdk.yml` commit

> You'll need to remove the `pandas`, `numpy`, `pyhdk` and `s3fs` dependencies from this file. In order to ensure that you have removed all of the necessary dependencies,

Reviewer: ```suggestion
You'll need to comment out the `pandas`, `numpy`, `pyhdk` and `s3fs` dependencies from this file. In order to ensure that you have removed all of the necessary dependencies,
```
Should this be replaced?

Also, can we make this sentence more generic to not specify the concrete packages? Because they may change over time and we should always remember to update this phrase to keep the packages in sync.

Author: Done!

check the patch introduced by the PR for the previous release version. After making these changes, commit the file with the commit message "Remove dependencies from env_hdk.yml."

### `setup.py` commit

> You'll need to replace the following list entries with empty lists in `setup.py`:

Reviewer: ```suggestion
You'll need to replace the following list entries with empty lists in `setup.py`:

```

* `dask_deps`
* `ray_deps`
* `unidist_deps`
* `spreadsheet_deps`

> * `sql_deps`

Reviewer 3: Removed dependencies
```suggestion
```

Reviewer: Is this addressed?
